File: app/src/test/java/com/example/themoviedb/PopularPeoplePresenterTest.java
package com.example.themoviedb;

import com.example.themoviedb.popular_people_screen.popular_people_model.AsyncResponseInterface;
import com.example.themoviedb.popular_people_screen.popular_people_model.PopularPeople;
import com.example.themoviedb.popular_people_screen.popular_people_model.PopularPeopleModelInterface;
import com.example.themoviedb.popular_people_screen.popular_people_presenter.PopularPeoplePresenter;
import com.example.themoviedb.popular_people_screen.popular_people_view.PopularPeopleViewInterface;

import org.junit.Before;
import org.junit.Test;
import org.mockito.ArgumentCaptor;
import org.mockito.Captor;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.List;

import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class PopularPeoplePresenterTest {

    PopularPeoplePresenter popularPeoplePresenter;
    @Mock
    private PopularPeopleViewInterface popularPeopleViewInterface;

    @Mock
    private PopularPeopleModelInterface popularPeopleModelInterface;

    @Mock
    private AsyncResponseInterface asyncResponseInterface;
    @Before
    public void setUp (){
        MockitoAnnotations.initMocks(this);
        popularPeoplePresenter = new PopularPeoplePresenter(popularPeopleViewInterface,popularPeopleModelInterface);
    }

    @Captor
    private ArgumentCaptor<AsyncResponseInterface> asyncResponseInterfaceArgumentCaptor;

    @Test
    public void callFetchingData (){
        PopularPeople popularPeopleList =  new PopularPeople();
        popularPeoplePresenter.callFetchingData( "https://api.themoviedb.org/3/person/popular?api_key=e6f20f39139b1f5a2be132cbaaa9ce43&page=1");
    }
}

File: app/src/test/java/com/example/themoviedb/PopularPeoplePresenterTest.kt
package com.example.themoviedb

import com.example.themoviedb.popular_people_screen.popular_people_model.PopularPeople
import com.example.themoviedb.popular_people_screen.popular_people_model.PopularPeopleModelInterface
import com.example.themoviedb.popular_people_screen.popular_people_presenter.PopularPeoplePresenter
import com.example.themoviedb.popular_people_screen.popular_people_view.PopularPeopleViewInterface
import com.nhaarman.mockito_kotlin.*
import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertTrue
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.MockitoAnnotations
import java.lang.AssertionError

class PopularPeoplePresenterTest {
    internal var popularPeoplePresenter: PopularPeoplePresenter? = null

    @Mock
    internal lateinit var popularPeopleViewInterface : PopularPeopleViewInterface

    @Mock
    internal lateinit var popularPeopleModelInterface: PopularPeopleModelInterface

    @Before
    fun setUp ()  {
        MockitoAnnotations.initMocks(this)
        popularPeoplePresenter = PopularPeoplePresenter(popularPeopleViewInterface,popularPeopleModelInterface)
    }

    @Test
    fun callScrollFunctionTest() {
        val expectedPageNumber = 2

        popularPeoplePresenter?.callScrollingFun()

        whenever(popularPeoplePresenter?.searchState != true )

        assertEquals(expectedPageNumber, popularPeoplePresenter?.page)
    }

    @Test
    fun callSearchScrollFunctionTest() {
        val expectedPageNumber = 2

        popularPeoplePresenter?.callScrollingFun()

        whenever(popularPeoplePresenter?.searchState == true )

        assertEquals(expectedPageNumber,  popularPeoplePresenter?.page)
    }

    @Test
    fun clearListTest() {
        popularPeoplePresenter?.callFetchingData()
        var size = popularPeopleViewInterface.getList()?.size
         size = 20
        val expectedSize = 0
        var sizeAfterClear :Int

        if (size != null) {
            if (size > 0){
                popularPeoplePresenter?.clearList()
            }

        }
        sizeAfterClear = popularPeopleViewInterface.getList()?.size!!
        assertEquals(expectedSize, sizeAfterClear)
    }

//    @Test
//    fun callSwipeFunctionTest() {
//
//        val expectedPageNumber = 1
//        var pageNumber = popularPeoplePresenter?.page
//        pageNumber = 2
//
////        popularPeoplePresenter?.callFetchingData()
//        popularPeoplePresenter?.callSwipeFun()
//
////        this.clearListTest()
//
////        whenever(popularPeoplePresenter?.searchState != true ).thenReturn(true)
//            assertEquals(expectedPageNumber, popularPeoplePresenter?.page)
//    }
}
